tools/commentMonitor/main.go

```diff
@@ -32,6 +32,7 @@ type commentMonitorConfig struct {
 	verifyUserDisabled bool
 	eventMapFilePath   string
 	whSecretFilePath   string
+	commandPrefixes    string
```

Comment:
Better to declare `commandPrefixes` as `[]string`.
In this way, we don't need to split it on every request during `checkCommandPrefix`.

May `StringsVar` in kingpin help.

Reply:
Yes, that's better but what do you think of putting the prefixes in the eventmap file like this, might have to rename `eventmap.yml` to something like `config.yml`:

```yaml
prefixes:
  - /prombench
  - /funcbench
eventmap:
  - event_type: event2
    regex_string: some_regex
    comment_template: |
      Dummy funcbench start
  - event_type: event1
    regex_string: some_regex
    comment_template: |
      Cool
    label: coolbench
```

Reply:
It makes sense to rename. But do you know the reason why we `loadConfig` on every request?
If it's a static part, we'd better preprocess it as much as possible, and not mix it with dynamical part, such as validation, extracting args.

Reply:
@nevill  : https://github.com/prometheus/test-infra/pull/307#discussion_r350567334

Reply:
I would argue that, even eventmap.yml is cached, the function call to unmarshal is not.
Plus, if we use config.yml, should we put other configs webhooksecret / port etc into it either?
I think we'd keep prefixes as the input parameters from cli.

Reply:
Okay, will update it for multiple `--command-prefix` (possibly with some short flag) then for now, can work on the other issues in another PR if needed.

> if we use config.yml, should we put other configs webhooksecret / port etc into it either?

No I think the prefix is related to the github comment command and the regex, so they make sense to be somewhere near where the regexes for the command is defined. (currently `eventmap.yml`) but other things like webhook secret and port are specific to how we run comment-monitor. Maybe `config.yml` is not the best name for the kind of file I am trying to describe, we can still keep `eventmap.yml` but if we include prefixes in it, it won't be a good name for the content of the file imo.

Reply:
if we have a small number of cli args I would say to avoid the need of a config file.

on a similar note do we even need the eventmap parsing at all?

Reply:
Added the changes, 

>  do we even need the eventmap parsing at all?

It think we'll be adding more commands so I guess having a file for storing the event types and regex is a good idea. About loading it on each http request, I think we don't really need that unless we're making changes to the file often and we can't afford any downtime for commentMonitor which currently is not the case. Were you suggesting something else?

Reply:
> I guess having a file for storing the event types and regex is a good idea.

Could you mention few benefits of keeping this file?

I can't think of anything important to be worth keeping it.
IIRC it was needed only because of how GH actions work, but now that the tool doesn't run inside GH actions don't see a reason to keep it.

I would say up to 10-15 cli args it is absolutely fine and don't require having a config file.

Reply:
As per this PR (https://github.com/prometheus/test-infra/pull/207) we were just using the [`regex` from the command line](https://github.com/nikita-mk/prombench/blob/f918ceb908bf7c00b00756169fa94d8bc6792ce3/cmd/tools/commentMonitor/main.go#L23) when using it inside github actions and the command template was set via env var later on.

But as we move away from github actions for comment parsing needs, we need to specify three things for each event, i.e `event_type`,`regex_string` and `comment_template`, now `label`(after #381 ) I think a file to manage this is simpler than having to pass these long comment strings as env vars or command line arguments. When using github actions, we were using the `event.json` file which was basically a dump of the `IssueComment` event webhook response.

`eventmap.yml` used for comment-monitor is different:
https://github.com/prometheus/test-infra/blob/master/prombench/manifests/cluster-infra/7a_commentmonitor_configmap_noparse.yaml

Reply:
ooh , yes I see what you mean now, these are too long for cli args, no other comments from my side, feel free to merge when you think it is ready